src/prediction_service_utils.cpp

```diff
@@ -72,13 +72,21 @@ Status getModelInstance(ovms::ModelManager& manager,
         if (modelInstance == nullptr) {
             return StatusCode::MODEL_VERSION_MISSING;
         }
-    } else {
+
+        return modelInstance->waitForLoaded(WAIT_FOR_MODEL_LOADED_TIMEOUT_MS, modelInstanceUnloadGuardPtr);
+    } 
+    
+    Status status = StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE;
+    auto retries = 5;
+    while (status == StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE && retries--) {
         modelInstance = model->getDefaultModelInstance();
         if (modelInstance == nullptr) {
             return StatusCode::MODEL_VERSION_MISSING;
         }
+        status = modelInstance->waitForLoaded(WAIT_FOR_MODEL_LOADED_TIMEOUT_MS, modelInstanceUnloadGuardPtr);
```

Comment:
Maybe add additional log info that retries failed? This is unlikely situation but in case of error we will have additional info for debug.

Reply:
added